src/Sylius/Bundle/ShopBundle/Grid/Account/OrderGrid.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ShopBundle\Grid\Account;
+
+use Sylius\Bundle\GridBundle\Builder\ActionGroup\ItemActionGroup;
+use Sylius\Bundle\GridBundle\Builder\Action\Action;
+use Sylius\Bundle\GridBundle\Builder\Action\ShowAction;
+use Sylius\Bundle\GridBundle\Builder\Field\DateTimeField;
+use Sylius\Bundle\GridBundle\Builder\Field\TwigField;
+use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Bundle\GridBundle\Grid\AbstractGrid;
+use Sylius\Bundle\GridBundle\Grid\ResourceAwareGridInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Customer\Context\CustomerContextInterface;
+
+final class OrderGrid extends AbstractGrid implements ResourceAwareGridInterface
+{
+    public function __construct(
+        private string $resourceClass,
+        private CustomerContextInterface $customerContext,
+        private ChannelContextInterface $channelContext,
+    ) {
+    }
+
+    public static function getName(): string
+    {
+        return 'sylius_shop_account_order';
+    }
+
+    public function buildGrid(GridBuilderInterface $gridBuilder): void
+    {
+        $gridBuilder
+            ->setRepositoryMethod('createByCustomerAndChannelIdQueryBuilder', [
+                $this->customerContext->getCustomer()?->getId(),
```

Comment:
> This kind of situation will be repeating with yaml->php config conversion as it was not possible to write complete code in yaml with null and strict type checks.

That's another case of this, query without customer is meaningless, but was allowed due no possibility to handle it in `yaml`

Reply:
The question here is, should we throw a `BadRequestException` or a not `Unauthorized` if we can't find the user.

Reply:
In this case the `CustomerContext` already seems to return only a `CustomerObject` from its signiture.

Reply:
IMO assert's `\InvalidArgumentException` is good enough, it's up to code, that stands before grid to check for authentication.

With this way (with assert) we're conceptually just ensuring "arguments"/global state validness, rather than working/handling request. 

Plus it feels a bit weird to throw http exception from "support" code like configuration file.

--- 
> In this case the CustomerContext already seems to return only a CustomerObject from its signiture.

Channel yes, but customer is nullable
https://github.com/Sylius/Sylius/blob/a88c39a9de5a3650fc814de51f21bad37a0224d9/src/Sylius/Component/Customer/Context/CustomerContextInterface.php#L20